src/ray/protobuf/common.proto

```diff
@@ -536,6 +542,35 @@ enum TaskStatus {
   RUNNING = 6;
 }
 
+enum SchedulingState {
+  // TODO(sang): Support more fine-grained states. E.g., split WAITING into
+  // WAITING_FOR_RESOURCES and WAITING_FOR_OBJECTS_LOCAL.
+  // Waiting to be scheduled.
+  WAITING = 0;
+  // Waiting for ray worker process to start.
+  WAITING_FOR_WORKERS = 1;
+  // Scheduling has been cancelled.
+  CANCELLED = 2;
+  // Scheduling is infeasible.
+  INFEASIBLE = 3;
+}
+
+message SchedulingInformation {
+  // The lease id of this scheduling request.
+  // Lease id is set at the client side. For tasks,
+  // lease id is a randomly assigned id. For actor scheduling,
+  // lease id is equivalent to the task id of the actor creation request.
+  // Actor tasks are not scheduled by a scheduler.
+  bytes lease_id = 1;
```

Comment:
do we need more information about lease (like the node id initiate the lease). 
could be done in follow up PR though